### PR TITLE
Bug 1874713: deployment: don't panic when applying deployment fails

### PR DESCRIPTION
**Pull request by stlaz**
In case there was an error during `ApplyDeployment`, the deployment would be `nil`, the controller would panic never to recover again.

/assign @mfojtik

### CODE REVIEW

File: pkg/controllers/deployment/deployment_controller.go [hunk 1]
```diff
@@ -362,3 +352,33 @@ func (c *deploymentController) ensureBootstrappedOAuthClients(ctx context.Contex
...
+
+// updateOperatorDeploymentInfo updates the operator's Status .ReadyReplicas, .Generation and the
+// .Generetions field with data about the oauth-server deployment
+func (c *deploymentController) updateOperatorDeploymentInfo(
+	ctx context.Context,
+	syncContext factory.SyncContext,
+	operatorConfig *operatorv1.Authentication,
+	deployment *appsv1.Deployment,
+) error {
+	operatorStatusOutdated := operatorConfig.Status.ObservedGeneration != operatorConfig.Generation || operatorConfig.Status.ReadyReplicas != deployment.Status.UpdatedReplicas
```
Comment (p0lyn0mial) [inline pkg/controllers/deployment/deployment_controller.go:364]: why we are not using [the generic workload controller](https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/controller/workload/workload.go) for this deployment? 

The generic controller is used to deploy the openshift-apiserver and the new oauth-apiserver

**Comment by sttts** [comment]
/lgtm

@p0lyn0mial your question is a good one. Would like to hear the answer. Let's fix the panic first though.

**Comment by stlaz** [comment]
We don't use it because it only came after we started using the current logic. I bet there are side effects that would need to be solved (setting the operand version, for example, there might be more).

It's (probably) not impossible to use the generic workload controller, I just wanted to have this fixed fast.

**Comment by mfojtik** [comment]
/lgtm

@stlaz sounds like a tech debt Jira or bugzilla (your choice) :-) not urgent.

**Pull request merged**